src/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java

```diff
@@ -158,6 +172,26 @@ public int getIntrinsicHeight() {
       this.intrinsicHeight = EMOJI_RAW_HEIGHT * decodeScale;
     }
 
+    private Bitmap getEmojiBitmap() {
+      Bitmap singleEmoji = Bitmap.createBitmap((int) intrinsicWidth/2, (int) intrinsicHeight/2, Bitmap.Config.ARGB_8888);
+
+      final int row = info.getIndex() / EMOJI_PER_ROW;
+      final int rowIndex = info.getIndex() % EMOJI_PER_ROW;
+
+      Rect desRect = new Rect(0, 0, (int) intrinsicWidth, (int) intrinsicWidth);
+      Rect srcRect = new Rect((int)(rowIndex * intrinsicWidth),
+        (int)(row * intrinsicHeight + row * verticalPad)+1,
+        (int)(((rowIndex + 1) * intrinsicWidth)-1),
+        (int)((row + 1) * intrinsicHeight + row * verticalPad)-1);
+
+      Canvas canvas = new Canvas(singleEmoji);
+      canvas.scale(0.5f, 0.5f);
```

Comment:
Where does the 0.5 scale comes from? Why the emoji returned is half the size of intrinsicWidth and intrinsicHeight?

Reply:
the original size of each emoji is 64x64 which is far too big, if it's rendered on the map. So we need to scale the cut out emoji down.

Reply:
But EmojiProvider itself and getEmojiBitmap has nothing to do with the map. Maybe add an argument to set bitmap scale and pass 0.5 from the map code here?

Reply:
good idea +1